### ISSUE TITLE
Parallelize nondeterministic WolframModel on GPU

### ISSUE THREAD

**Issue by maxitg** [maxitg/SetReplace#354]
Requires: #351.

### The problem
This is similar to #352, but for GPUs rather than CPUs. We need both because some users might not have access to a GPU, especially if we don't support all of them.

This issue specifically is for parallelization of a nondeterministic evolution `"EventOrderingFunction" -> "Any"` as defined in #351. There is another issue #353 for parallelizing deterministic evolution.

### Additional context
Feel free to create subtasks for different GPU frameworks.
